src/main/java/window/FileSelector.java

```diff
@@ -53,10 +61,52 @@ public static FileChooser getInstance() {
      * @return The file we want to parse.
      */
     public static File showOpenDialog(Window ownerWindow) {
+        FileSystem fileSystem = new FileSystem();
+        logger = new LoggerFactory(fileSystem).createLogger(FileSelector.class);
+        String currentDir = openDirectory();
+        if (!currentDir.equals("")) {
+            File file = new File(currentDir);
+            getInstance().setInitialDirectory(file);
+        }
         File chosenFile = getInstance().showOpenDialog(ownerWindow);
+        logger.info("Selected file: " + chosenFile.getName());
         if (chosenFile != null) {
-            lastDir.setValue(chosenFile.getParentFile());
+            saveDirectory(chosenFile.getParentFile().getAbsolutePath());
         }
         return chosenFile;
     }
+
+    /**
+     * Finds the last used directory from which a file was chosen.
+     * @return the path of the directory.
+     */
+    private static String openDirectory() {
```

Comment:
This method name suggests this method will open a directory while it returns a directory path. Consider refactoring it.